Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -290,8 +290,11 @@ public InterpreterScope(AnalysisValue av, InterpreterScope outerScope)
             _nodeScopes.Clear();
         }
 
-        public virtual IAnalysisSet AddNodeValue(Node node, IAnalysisSet variable) {
-            return _nodeValues[node] = variable;
+        public virtual IAnalysisSet AddNodeValue(Node node, NodeValueKind kind, IAnalysisSet variable) {
+            NodeValue next;
+            _nodeValues.TryGetValue(node, out next);
+            _nodeValues[node] = new NodeValue(kind, variable, next);
```

Comment:
Correct, at most we'll currently end up with 2.  That open happens when we have a subclass of dict.  In NTVS I did something similar but that still had a Dictionary<TKey, TValue> for storing these so the kind was encoded in a struct with the node.  Because this has switched to AnalysisDictionary (presumably to fix a thread safety issue at some point) I didn't want to allocate a reference on all of the gets.  But we're bounded here by the number of NodeValueKind's, and even then the kinds would need to be started being used across many nodes to collide.